contrib/util/deidentification/deidentification.php

```diff
@@ -0,0 +1,414 @@
+//which facility the user has access too.
+function deIdFacilityTable($con){
+
+
+    removeColumn($con, "facility", "phone", "(000) 000-0000");
+    removeColumn($con, "facility", "street", "123 Somewhere");
+    removeColumn($con, "facility", "federal_ein", "123456789");
+    removeColumn($con, "facility", "attn", "Office Person");
+    removeColumn($con, "facility", "facility_npi", "987654321");
+
+    $query = "select * from facility";
+    $result = mysqli_query($con, $query);
+    while($row = mysqli_fetch_array($result)){
+
+        $string = "update facility set 
+          
+              `name`    = 'Facility_{$row['id']}',
+              `phone`   = '(000) 000-0000'
+    
+            where `id` = {$row['id']}";
+
+        mysqli_query($con, $string) or print "Error altering facility table \n";
+        $string = '';
+
+
+
+    }
+
+    echo "Successfully deid'ed Facility Table";
+
+}
+
+
+
+//
+function deIdUsersTable($con){
+
+
+    removeColumn($con, "users", "federaltaxid", "123456789");
+    removeColumn($con, "users", "federaldrugid", "D000-1");
+//    removeColumn($con, "users", "facility", "Service Facility");
+    removeColumn($con, "users", "email", "doc@home.com");
+    removeColumn($con, "users", "organization", "");
+    removeColumn($con, "users", "npi", "");
+    removeColumn($con, "users", "street", "");
+    removeColumn($con, "users", "streetb", "");
+    removeColumn($con, "users", "city", "");
+    removeColumn($con, "users", "state", "");
+    removeColumn($con, "users", "zip", "");
+    removeColumn($con, "users", "phone", "");
+    removeColumn($con, "users", "phonew1", "");
+    removeColumn($con, "users", "phonew2", "");
+    removeColumn($con, "users", "fax", "");
+    removeColumn($con, "users", "phonecell", "");
+    removeColumn($con, "users", "info", "");
+
+
+    $query = "select * from users";
+    $result = mysqli_query($con, $query);
+
+
+
+    while($row = mysqli_fetch_array($result)){
+        $string = "update users set ";
+
+       if(strpos($row['newcrop_user_role'], 'doctor') !==false){
+
+           $string .= "fname = 'Doctor.{$row['id']}', 
+                       lname = 'Doctor.{$row['id']}' ";
+
+
+       }else if(strpos($row['newcrop_user_role'], 'nurse') !==false){
+
+           $string .= "fname = 'Nurse.{$row['id']}', 
+                       lname = 'Nurse.{$row['id']}' ";
+
+       }else{
+
+           $string .= "fname = 'noNewCrop', 
+                       lname = 'Nurse{$row['id']}'";
+
+       }
+        $string .= " where `id` = {$row['id']} ";
+        mysqli_query($con, $string) or print "Error altering users table \n";
+        //$string = '';
+
+    }
+
+    echo "successfuly altered user table \n ";
+}
+
+//Clears most forms.  User must verify that this function handles all text fields that might hold personal identifying information
+function deIdForms($con){
+
+    removeColumn($con, "form_physical_exam", "comments", "no comment, talk to my lawyer");
```

Comment:
Nicer might be to look for the replaced names from patient_Data, which could leave you with realistic seeming data that is (mostly) de-identified.  Unfortunately, few people use the soap form or any of the others we have.  If those forms don't exist in the database, this breaks.